### PR TITLE
Make the ExecutionError readable for the run_check case

### PR DESCRIPTION
**Pull request by Emantor**
If you use the run_check function and get an ExecutionError, it will only contain the command that was run, but not the stdout or stderr. Add those and refactor run_check into the mixin.

### CODE REVIEW

**Comment by Emantor** [comment]
Rebased and added tests

File: labgrid/driver/commandmixin.py [hunk 1]
```diff
@@ -21,4 +21,22 @@ def wait_for(self, cmd, pattern, timeout=30.0, sleepduration=1):
         while not any(pattern in s for s in self.run_check(cmd)) and not timeout.expired:
             sleep(sleepduration)
         if timeout.expired:
-            raise ExecutionError("Wait timeout expired")
+            raise ExecutionError("Wait timeout expired", '')
+
+    @Driver.check_active
+    @step(args=['cmd'], result=True)
+    def run_check(self, cmd: str, timeout=30):
```
Comment (jluebbe) [inline labgrid/driver/commandmixin.py]: This removes the _run_check method, which can be used internally during activation and will break _put_ssh_key of the ShellDriver.

**Comment by Emantor** [comment]
Fixed the removal of the _run_check function.

**Comment by Emantor** [comment]
Moved _run_check into the mixin.

**Pull request merged**